### PR TITLE
Enable all the compiler checks

### PR DESCRIPTION
**Pull request by fredj**

### CODE REVIEW

**Comment by gberaudo** [comment]
Nice simplification: +1.

File: config/example.json [hunk 1]
```diff
@@ -29,41 +29,7 @@
...
-      "suspiciousCode",
-      "strictModuleDepCheck",
-      "typeInvalidation",
-      "undefinedNames",
-      "undefinedVars",
-      "uselessCode",
-      "visibility"
-    ],
-    "jscomp_off": [
-      "unknownDefines"
```
Comment (elemoine) [inline config/example.json]: I am not sure about this one. There was a good reason for turning off `unknownDefines`. I need to remember what it was.

Comment (fredj) [inline config/example.json]: And this config file is missing the others `jscomp_off` checks

**Comment by fredj** [comment]
testing now to compile all the examples with:
```for f in examples/*.js; do make build/compiled-examples/${f:9:-3}.combined.js; done```

**Comment by elemoine** [comment]
Thanks!

**Comment by fredj** [comment]
`unknownDefines` is necessary for the examples with the `NOCOMPILE` header

**Comment by elemoine** [comment]
> unknownDefines is necessary for the examples with the NOCOMPILE header

Not only I think. Without that flag set to off I guess the compiler will complain on any example that does not pull `goog.json` (because the `goog.json.USE_NATIVE_JSON` is unknown to the compiler if `goog/json/json.js` is not sent to the compiler).

**Comment by fredj** [comment]
> Not only I think. Without that flag set to off I guess the compiler will complain on any example that does not pull `goog.json` (because the `goog.json.USE_NATIVE_JSON` is unknown to the compiler if `goog/json/json.js` is not sent to the compiler).

Correct, I will add `unknownDefines` to `jscomp_off`

**Comment by fredj** [comment]
> `unknownDefines` is necessary for the examples with the `NOCOMPILE` header

`unknownDefines` != `undefinedVars` !

**Comment by elemoine** [comment]
Thanks @fredj. Please merge when Travis is done.

**Pull request merged**